src/hooks/use-load-topic-data.ts

```diff
@@ -1,26 +1,74 @@
 import * as React from 'react'
 import groq from 'groq'
 import {sanityClient} from '@/utils/sanity-client'
+import {topicExtractor} from '@/utils/search/topic-extractor'
+import {first} from 'lodash'
+import {loadTag} from '@/lib/tags'
+
+function useLoadTopicData(
+  initialTopicGraphqlData: any,
+  initialTopicSanityData: any,
+  searchState: any,
+) {
+  const [topicGraphqlData, setTopicGraphqlData] = React.useState<any>(
+    initialTopicGraphqlData,
+  )
+  const [topicSanityData, setTopicSanityData] = React.useState<any>(
+    initialTopicSanityData,
+  )
+  const [loading, setLoading] = React.useState<boolean>(false)
+
+  const selectedTopics: any = topicExtractor(searchState)
+  const newTopic =
+    selectedTopics.length === 1 ? first<string>(selectedTopics) : false
 
-function useLoadTopicData(topic: string, initialData: any) {
-  const [topicData, setData] = React.useState<any>(initialData)
-  const [isLoading, setLoading] = React.useState<boolean>(false)
   React.useEffect(() => {
+    const controller = new AbortController()
+    const signal = controller.signal
+
     async function getData() {
       setLoading(true)
-      await sanityClient
-        .fetch(topicQuery, {
-          slug: topic,
-        })
-        .then((data) => {
-          setData(data)
-          setLoading(false)
-        })
+      try {
+        const sanityData = await sanityClient.fetch(
+          topicQuery,
+          {
+            slug: newTopic,
+          },
+          {signal},
+        )
+
+        if (sanityData) {
+          setTopicSanityData(sanityData)
+          setTopicGraphqlData({name: sanityData.slug})
+        }
+
+        if (!sanityData && newTopic) {
+          const graphqlData = await loadTag(newTopic)
+          setTopicSanityData(null)
+          setTopicGraphqlData(graphqlData)
+        }
+      } catch (error: any) {
+        if (error.name !== 'AbortError') {
+          console.error('Error fetching topic data: ', error)
+        }
+      } finally {
+        setLoading(false)
+      }
+    }
+
+    if (newTopic) {
+      getData()
+    } else {
+      setTopicSanityData(null)
+      setTopicGraphqlData({name: ''})
```

Comment:
A topic is always expected. A next step could be to handle null in the template. I don't like having to set this empty string.

I wanted to stay within the scope of fixing the search page bug in this PR